Firestore/core/src/model/document_key.h

```diff
@@ -57,6 +59,9 @@ class DocumentKey {
   /** Creates and returns a new document key with the given segments. */
   static DocumentKey FromSegments(std::initializer_list<std::string> list);
 
+  /** Returns a DocumentKey from a fully qualified resource name. */
+  static DocumentKey FromName(absl::string_view name);
```

Comment:
Hmm, this has the potential for the same problem as in `ResourcePath`. It should be okay as long as the code in the public part doesn't come to rely on this function, but that seems brittle.

OTOH, looking at how this is used, replacing the argument with a `const std::string&` would force an undesirable conversion. Unfortunately, the only two alternatives I can think of are unatrractive:
* make this function accept a char pointer and a size;
* make this function private and create a class that is a friend of this class and access `FromName` through it.

What do you think?

Reply:
I updated the method to take a std::string since it is used in the same code path that also uses the DatabaseId parser (which I reverted back to its original code).